pkg/system/system.go

```diff
@@ -1129,14 +1129,6 @@ func GetDesiredDBImage(sys *nbv1.NooBaa, currentImage string) string {
 		return psql12Image
 	}
 
-	if sys.Spec.DBImage != nil {
```

Comment:
This should not be removed. after removal you will always get the hardcoded image

Reply:
Restoring it and adding check for desired image in `UpgradeDB`